packages/mattermost-redux/src/selectors/entities/posts.ts

```diff
@@ -63,16 +65,27 @@ export function getReactionsForPosts(state: GlobalState): RelationOneToOne<Post,
     return state.entities.posts.reactions;
 }
 
-export function makeGetReactionsForPost(): (state: GlobalState, postId: $ID<Post>) => {
-    [x: string]: Reaction;
-} | undefined | null {
-    return createSelector('makeGetReactionsForPost', getReactionsForPosts, (state: GlobalState, postId: string) => postId, (reactions, postId) => {
-        if (reactions[postId]) {
-            return reactions[postId];
-        }
+export function makeGetReactionsForPost(): (state: GlobalState, postId: Post['id']) => Record<string, Reaction> | undefined | null {
+    return createSelector(
+        'makeGetReactionsForPost',
+        getEmojiMap,
+        getReactionsForPosts,
+        (_: GlobalState, postId: string) => postId,
+        (emojiMap, reactions, postId) => {
+            if (reactions[postId]) {
```

Comment:
Minor nit, we could do an early return if `!reactions[postId]` to avoid indenting as much of this

Reply:
✅  Updated as per request/discussion